packages/edit-site/src/components/sidebar-navigation-screen-global-styles/index.js

```diff
@@ -39,30 +38,26 @@ export default function SidebarNavigationScreenGlobalStyles() {
 		isLoading: isLoadingRevisions,
 		revisionsCount,
 	} = useGlobalStylesRevisions();
-	const { openGeneralSidebar } = useDispatch( editSiteStore );
+	const editorCanvasContainerView = useSelect(
+		( select ) =>
+			unlock( select( editSiteStore ) ).getEditorCanvasContainerView(),
+		[]
+	);
 	const { setEditorCanvasContainerView } = unlock(
 		useDispatch( editSiteStore )
 	);
-	const { set: setPreference } = useDispatch( preferencesStore );
-
-	const openGlobalStyles = useCallback( async () => {
-		history.navigate( addQueryArgs( path, { canvas: 'edit' } ), {
-			transition: 'canvas-mode-edit-transition',
-		} );
-		return Promise.all( [
-			setPreference( 'core', 'distractionFree', false ),
-			openGeneralSidebar( 'edit-site/global-styles' ),
-		] );
-	}, [ path, history, openGeneralSidebar, setPreference ] );
-
-	const openRevisions = useCallback( async () => {
-		await openGlobalStyles();
-		// Open the global styles revisions once the canvas mode is set to edit,
-		// and the global styles sidebar is open. The global styles UI is responsible
-		// for redirecting to the revisions screen once the editor canvas container
-		// has been set to 'global-styles-revisions'.
-		setEditorCanvasContainerView( 'global-styles-revisions' );
-	}, [ openGlobalStyles, setEditorCanvasContainerView ] );
+	const openRevisions = useCallback( () => {
+		if ( 'style-book' === editorCanvasContainerView ) {
```

Comment:
Note: this PR has exposed the clunky way we're loading the editor canvas views. There are significant "flashes" when navigating between revisions and the main styles panel in view mode. 

I'm not sure of an alternative that maintains current functionality, probably to create a bespoke stylebook component for the view mode, which is duplication of effort and maintenance. 

Or better handling of transitions between editor views using CSS or something. 

All this is keeping the pot simmering while a better presentation layer to load global-styles related editor views is worked out.